client/src/main/scala/com/dwolla/cloudflare/CloudflareClient.scala

```diff
@@ -1,10 +1,36 @@
 package com.dwolla.cloudflare
 
+import java.io.Closeable
+
 import cats.Monad
+import cats.effect.Async
 import com.dwolla.cloudflare.clients.{AccountsClient, DnsRecordClient, RateLimitClient}
 
-class CloudflareClient[F[_] : Monad](executor: CloudflareApiExecutor[F]) {
-  lazy val Accounts = new AccountsClient(executor)
-  lazy val RateLimits = new RateLimitClient(executor)
-  lazy val Dns = new DnsRecordClient(executor)
+import scala.concurrent.{ExecutionContext, Future}
+import scala.language.higherKinds
+
+trait CloudflareClient[F[_]] {
+  def Accounts: AccountsClient[F]
+  def RateLimits: RateLimitClient[F]
+  def Dns: DnsRecordClient[F]
 }
+
+class FutureCloudflareClient(authorization: CloudflareAuthorization)(implicit ec: ExecutionContext, implicit val evidence: Monad[Future]) extends CloudflareClient[Future] with Closeable {
```

Comment:
@bpholt I'm sure there are easier/better ways to do this, would love your help. Was trying to keep anyone using this lib from having to create an executor and pass it in, and just create a client that wrapped everything else.